app/src/main/shared/Helpers.ts

```diff
@@ -36,5 +39,24 @@ export const helpers = {
             obj[key] = value;
         });
         return obj;
+    },
+    ingestQueryStringAndReturnResult<TModel>(): Result | void {
+
```

Comment:
Juri v rightly pointed out that we were reading the query string without removing it from the window location, meaning a user could GET the page and get an outdated response back. Also following the proposed pattern of the form itself displaying the success/error message without passing it to the global notification store. So this parses the query string, removes from the window location, and returns the result for a component to do what it will with it.